src/libstore/build.cc

```diff
@@ -1629,6 +1629,19 @@ void DerivationGoal::buildDone()
            being valid. */
         registerOutputs();
 
+        if (settings.postBuildHook != "") {
+            printMsg(lvlChatty, format("executing post-build hook '%1%'")
```

Comment:
We should probably bump this up out of chatty, in to something higher so `nix-build` clients typically see it. Otherwise, there is a long pause between the build finishing and the step being considereh completed.

Reply:
You can use `Activity` for that.